compiler/src/swc.rs

```diff
@@ -0,0 +1,610 @@
+        wr: writer,
+      };
+      program.emit_with(&mut emitter).unwrap();
+    }
+    let src = String::from_utf8(buf).unwrap();
+    if source_map {
+      let mut buf = Vec::new();
+      self
+        .source_map
+        .build_source_map_from(&mut src_map_buf, None)
+        .to_writer(&mut buf)
+        .unwrap();
+      Ok((src, Some(String::from_utf8(buf).unwrap())))
+    } else {
+      Ok((src, None))
+    }
+  }
+}
+
+fn get_es_config(jsx: bool) -> EsConfig {
+  EsConfig {
+    class_private_methods: true,
+    class_private_props: true,
+    class_props: true,
+    dynamic_import: true,
+    export_default_from: true,
+    export_namespace_from: true,
+    import_meta: true,
+    jsx,
+    nullish_coalescing: true,
+    num_sep: true,
+    optional_chaining: true,
+    top_level_await: true,
+    ..EsConfig::default()
+  }
+}
+
+fn get_ts_config(tsx: bool) -> TsConfig {
+  TsConfig {
+    tsx,
+    decorators: true,
+    dynamic_import: true,
+    ..TsConfig::default()
+  }
+}
+
+fn get_syntax(source_type: &SourceType) -> Syntax {
+  match source_type {
+    SourceType::JavaScript => Syntax::Es(get_es_config(false)),
+    SourceType::JSX => Syntax::Es(get_es_config(true)),
+    SourceType::TypeScript => Syntax::Typescript(get_ts_config(false)),
+    SourceType::TSX => Syntax::Typescript(get_ts_config(true)),
+    _ => Syntax::Es(get_es_config(false)),
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use crate::aleph::VERSION;
+  use crate::import_map::ImportHashMap;
+  use crate::resolve::{DependencyDescriptor, Resolver, HASH_PLACEHOLDER};
+
+  fn t(specifer: &str, source: &str, bundling: bool) -> (String, Rc<RefCell<Resolver>>) {
+    let module = ParsedModule::parse(specifer, source, None).expect("could not parse module");
+    let resolver = Rc::new(RefCell::new(Resolver::new(
+      specifer,
+      ImportHashMap::default(),
+      None,
+      bundling,
+      vec![],
+    )));
+    let (code, _) = module
+      .transpile(resolver.clone(), &EmitOptions::default())
+      .expect("could not transpile module");
+    (code, resolver)
+  }
+
+  #[test]
+  fn test_transpile_ts() {
+    let source = r#"
+      enum D {
+        A,
+        B,
+        C,
+      }
+
+      function enumerable(value: boolean) {
+        return function (
+          _target: any,
+          _propertyKey: string,
+          descriptor: PropertyDescriptor,
+        ) {
+          descriptor.enumerable = value;
+        };
+      }
+
+      export class A {
+        private b: string;
+        protected c: number = 1;
+        e: "foo";
+        constructor (public d = D.A) {
+          const e = "foo" as const;
+          this.e = e;
+        }
+        @enumerable(false)
+        bar() {}
+      }
+    "#;
+    let (code, _) = t("https://deno.land/x/mod.ts", source, false);
+    println!("{}", code);
+    assert!(code.contains("var D;\n(function(D) {\n"));
+    assert!(code.contains("_applyDecoratedDescriptor("));
```

Comment:
Along with the other tests, is there any reason for only checking `contains` of the output code? Is it because it would take a lot more lines writing the output with `r#`? There is most likely another way I think, but I am not that good with Rust so idk.

Reply:
`contains` is enough i think, no need to check full output, since swc' output (format) may change in the future